Tests/LootLockerTests/PlayMode/LeaderboardTest.cs

```diff
@@ -0,0 +1,495 @@
+            string leaderboardKey = "gl_leaderboard";
+            var createLeaderboardRequest = new CreateLootLockerLeaderboardRequest
+            {
+                name = "Global Leaderboard",
+                key = leaderboardKey,
+                direction_method = "ascending",
+                enable_game_api_writes = true,
+                has_metadata = true,
+                overwrite_score_on_submit = false,
+                type = "player"
+            };
+
+            bool leaderboardCreated = false;
+            bool leaderboardSuccess = false;
+            gameUnderTest.CreateLeaderboard(createLeaderboardRequest, (success, errorMessage, response) =>
+            {
+                if (success)
+                {
+                    leaderboardSuccess = true;
+
+                }
+                else
+                {
+                    Debug.Log(errorMessage);
+                    SetupFailed = true;
+                }
+                leaderboardCreated = true;
+
+            });
+            yield return new WaitUntil(() => leaderboardCreated);
+
+            Assert.IsTrue(leaderboardSuccess, "Failed to create leaderboard");
+
+            for (int i = 0; i < 10; i++)
+            {
+                bool guestSessionCompleted = false;
+                LootLockerSDKManager.StartGuestSession(Random.Range(100, 10000).ToString(), (response) =>
+                {
+                    guestSessionCompleted = true;
+
+                });
+                yield return new WaitUntil(() => guestSessionCompleted);
+
+                bool scoreSubmitted = false;
+                LootLockerSDKManager.SubmitScore("", Random.Range(5, 100), leaderboardKey, (response) =>
+                {
+                    scoreSubmitted = true;
+                });
+                yield return new WaitUntil(() => scoreSubmitted);
+
+            }
+            
+            //When
+            LootLockerGetScoreListResponse actualResponse = null;
+            bool scoreListCompleted = false;
+            LootLockerSDKManager.GetScoreList(leaderboardKey, 10, (response) =>
+            {
+                actualResponse = response;
+                scoreListCompleted = true;
+            });
+            yield return new WaitUntil(() => scoreListCompleted);
+
+            bool isSortedRight = true;
+            int lastScore = 0;
+
+            for(int i = 0; i < actualResponse.items.Length; i++)
+            {
+                if(i == 0)
+                {
+                    lastScore = actualResponse.items[i].score;
+                }
+                else
+                {
+                    if(lastScore > actualResponse.items[i].score)
+                    {
+                        isSortedRight = false;
+                    }
+                    lastScore = actualResponse.items[i].score;
+                }
+            }
+
+            //Then
+            Assert.IsTrue(isSortedRight, "Score list was not sorted ascending");
```

Comment:
If this test was to fail it would be incredibly annoying to debug, can you make it more explicit somehow?

Reply:
Removed test